core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -83,14 +85,16 @@ public SqlSelect(SqlParserPos pos,
                    SqlNode offset,
                    SqlNode fetch,
                    SqlNodeList hints) {
-    this(pos, keywordList, /*topN=*/ null, selectList, from, where, groupBy,
-        having, qualify, windowDecls, orderBy, offset, fetch, hints);
+    this(pos, keywordList, /*topN=*/ null, /*exceptExpression=*/ null, selectList,
```

Comment:
From the constructor definition, it looks like exceptExpression actually comes after selectList.